src/OptionsCompounderAave2.sol

```diff
@@ -271,57 +223,91 @@ contract OptionsCompounder is FlashLoanReceiverBase, Ownable {
             balanceOfUnderlyingToken
         );
         /* Swap underlying token to payment token (asset) */
-        // Question: Now there is an assumption that payment token is the strategy "want" token
-        // If we would like to have different want token, here is the place to querry strategy about it
-        // For the sake of tests it is not yet done
+        // Question: Here is some room for optimization. Instead of swapping all underlying tokens
+        // to payment tokens, we can swap necessary amount of payment tokens (totalAmount) and the rest
+        // underlying tokens can be swapped to the want token but swapper doesn't allow to put
+        // here amountOut (it is amountIn acceptable for swapBal). Is it worth to play with this ?
```

Comment:
worth at least thinking about optimizations, can discuss more in mtg, can special case when payment token == want token